src/lib/layout/status.rhizome.svelte

```diff
@@ -0,0 +1,91 @@
+<script lang="ts">
+  import Icon from "@iconify/svelte";
+  import { listen } from "@tauri-apps/api/event";
+  import { onMount } from "svelte";
+
+  /** The current status of the rhizome connection */
+  let rzStatus: string = "connecting";
+  /** Time since connect was called */
+  let rzMili: string = "0";
+
+  onMount(() => {
+    // Try to load the rhizome status from the session storage.
+    rzStatus = sessionStorage.getItem("rz-status") ?? "connecting";
+    rzMili = sessionStorage.getItem("rz-mili") ?? "0";
+
+    console.log(sessionStorage.getItem("rz-mili"));
+
+    listen("rz-con", (evt) => setStatus("connected", evt.payload as string));
+    listen("rz-dc", (evt) => setStatus("disconnected", evt.payload as string));
+    listen("rz-f", (evt) => setStatus("failed", evt.payload as string));
+  });
+
+  function setStatus(status: string, mili: string) { 
+    rzStatus = status; 
+    sessionStorage.setItem("rz-status", status); 
+
+    if (mili !== "null" && mili !== null) {
```

Comment:
are you sure we need to check that as this may never be null if the rust side is implemented properly